### PR TITLE
mysql: add write conflict error for latch and change the error code of ErrTxnTooLarge

### PR DESCRIPTION
**Pull request by jackysp**
Signed-off-by: Shuaipeng Yu <jackysp@gmail.com>

### What problem does this PR solve? 
Add write conflict error for latch and change the error code of ErrTxnTooLarge.

### CODE REVIEW

File: mysql/errcode.go [hunk 1]
```diff
@@ -941,6 +943,4 @@ const (
 	ErrRegionUnavailable  = 9005
 	ErrGCTooEarly         = 9006
 	ErrWriteConflict      = 9007
-
-	ErrTxnTooLarge = 9500
```
Comment (coocood) [inline mysql/errcode.go]: Why change this?

Comment (jackysp) [reply mysql/errcode.go]: It is not a related change. But ErrTxnTooLarge is a TiDB error, it should not in the TiKV/PD section.

Comment (jackysp) [reply mysql/errcode.go]: It is better to keep the TiDB errors starts from 8000 and TiKV/PD's starts from 9000.

Comment (jackysp) [reply mysql/errcode.go]: I've updated the title and the description.

**Comment by coocood** [comment]
LGTM

**Pull request merged**